rs/index/src/ivf/files/invalidated_doc_ids.rs

```diff
@@ -0,0 +1,140 @@
+use std::fs::OpenOptions;
+
+use anyhow::{anyhow, Result};
+use memmap2::MmapMut;
+
+pub struct InvalidatedDocIdsStorage {
+    mmap: MmapMut,
+    num_docs: usize,
+}
+
+const BITS_IN_BYTE: usize = 8;
+
+impl InvalidatedDocIdsStorage {
+    pub fn new(base_directory: &str, num_docs: usize) -> Result<Self> {
+        let file = OpenOptions::new()
+            .read(true)
+            .write(true)
+            .create(true)
+            .open(format!("{}/invalidated_docs", base_directory))?;
+
+        let bytes_needed = (num_docs + BITS_IN_BYTE - 1) / BITS_IN_BYTE;
+        file.set_len(bytes_needed as u64)?;
+
+        let mmap = unsafe { MmapMut::map_mut(&file)? };
+        Ok(Self { mmap, num_docs })
+    }
+
+    fn set(&mut self, doc_id: usize, invalid: bool) -> Result<()> {
```

Comment:
let's call the `id` in this file `point_id`, since we expect it to go from `0..num_docs`. Also make it u32